### DISCUSSION TITLE
Automatically assign available database public port

### DISCUSSION THREAD

**Discussion by mbeaudru** [coollabsio/coolify#2062] [Feature Requests]
👋 

Today, to make a database publicly available you have to specify a public port to expose it.

Problem is, it's not that easy to find available ports and the error message when a port is already used is quite "scary"

<img src="https://github.com/coollabsio/coolify/assets/8942633/193186f1-5647-4d37-9801-da21205acd29" width="300px" />

## Suggestion

Auto-assign available port when clicking on the "make the database publicly available" checkbox

I personally installed & used [get-port-cli](https://github.com/sindresorhus/get-port-cli) package on the server to overcome this issue easily, but anything like it would work I guess 👍

**Comment by puria** [comment]
@mbeaudru do you run it manually, before? I also want be able to deploy many same service on the same machine and be able to assign a random port, behind the reverse proxy.

I guess also related to https://github.com/coollabsio/coolify/discussions/4743 and #2290
and maybe easily can be solved to have some `docker-compose coolify magic` that do create a random integer between 1024 and 65535

**Reply by puria** [reply]
Seems that for temporary <3, just exposing 
```php 
use Illuminate\Support\Facades\Random;

$randomPort = Random::randomInt(1024, 65535);
```
should be enough for a rough patch, as is done with `Str::random`
